### PR TITLE
Fix the cmake packaging issue on macOS

### LINKED ISSUE #6025: Bundle step fails on the macOS docs workflow
**Description of the problem**

The macOS docs workflow is consistently failing on the Package GMT step (e.g. https://github.com/GenericMappingTools/gmt/runs/4244772423?check_suite_focus=true) with the error:
```
  /Applications/Xcode_13.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic:
  can't open file: @rpath/libgeos.3.10.1.dylib (No such file or directory)
```

So far, I have tried adding geos to [install-dependencies-macos.sh](https://github.com/GenericMappingTools/gmt/blob/master/ci/install-dependencies-macos.sh) and re-adding the `brew update` step before `brew install`, with no success.

@seisman, do you have any recommendations for fixing the workflow? Otherwise, I can switch from MacPorts to homebrew after the 6.3 release to test the setup locally.

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

The workflow fails to make the macOS bundle, with the following error message:
```
/Applications/Xcode_13.0.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic:
  can't open file: @rpath/libgeos.3.10.1.dylib (No such file or directory)
```
It's unclear why other libraries (like gdal) have abosulte paths, but the geos library has a relative path `@rpath/libgeos.3.10.1.dylib`. So the root cause of the error is still unclear to me.

A temporary workaround is passing extra RPATH (`/usr/local/lib` for homebrew) as the last parameter of the `fixup_bundle` function. 

Fixes #6025.

### CODE REVIEW

**Comment by seisman** [comment]
With the workaround in this PR, the workflow now can create macOS bundle correctly.

@meghanrjones Please check if it works for you (I assume that you're a Homebrew user)
@PaulWessel Please check if it breaks your building with macports.

**Comment by meghanrjones** [comment]
Thanks for working on this. `admin/build-release.sh -t` ran without errors for me using this PR.

**Comment by joa-quim** [comment]
Will the bundle with this build be usable for pygmt?

**Comment by PaulWessel** [comment]
I think that is a separate PR that we need to work on for both Julia and PyGMT.

**Comment by seisman** [comment]
@PaulWessel does this PR affects your bundle building with macports?

**Comment by PaulWessel** [comment]
Not sure, will try soon

**Comment by PaulWessel** [comment]
No errors on Mac ports. I have not actually tested the bundle yet, maybe on upcoming flight

**Comment by PaulWessel** [comment]
Build works, and running gmt coast with the installed bundle works for me on M1 macOS, macports..

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 name: Docs
 
 on:
-  # pull_request:
+  pull_request:
```
Comment (seisman) [inline .github/workflows/docs.yml:7]:
```suggestion
  # pull_request:
```

**Pull request merged**